ai/STA/Strategy/penalty_offense.py

```diff
@@ -32,17 +38,54 @@ def __init__(self, game_state):
                              Role.SECOND_DEFENCE: Pose.from_values(our_goal.position.x / 8, field.bottom * 2 / 3)}
 
         kicker = self.assigned_roles[Role.MIDDLE]
-        self.create_node(Role.MIDDLE, GoKick(game_state,
-                                             kicker,
-                                             kick_force=KickForce.HIGH,
-                                             forbidden_areas=[new_goal],
-                                             auto_update_target=True))
+        top_hole = self.game_state.field.their_goal.copy()
+        top_hole.y += self.game_state.field.goal_width / 2 - 2 * ROBOT_RADIUS
+        bot_hole = self.game_state.field.their_goal.copy()
+        bot_hole.y -= self.game_state.field.goal_width / 2
+
+        their_goal_to_ball = self.game_state.ball_position - bot_hole
+        go_behind_position = self.game_state.ball_position + their_goal_to_ball.unit * ROBOT_RADIUS * 2.0
+        go_behind_orientation = their_goal_to_ball.angle + math.pi
+
+        go_behind = self.create_node(Role.MIDDLE, GoToPosition(self.game_state,
+                                                               kicker,
+                                                               target=Pose(go_behind_position, go_behind_orientation),
+                                                               cruise_speed=1,
+                                                               ball_collision=False))
+        go_kick = self.create_node(Role.MIDDLE, GoKick(game_state,
+                                                 kicker,
+                                                 target=Pose(top_hole),
+                                                 kick_force=KickForce.HIGH,
+                                                 forbidden_areas=[new_goal]))
+
+        go_behind.connect_to(go_kick, when=self.goalkeeper_is_top_or_timeout)
+        go_kick.connect_to(go_behind, when=self.not_goalkeeper_is_top_or_timeout)
+        self.start_time = time.time()
 
         goalkeeper = self.assigned_roles[Role.GOALKEEPER]
         self.create_node(Role.GOALKEEPER, GoalKeeper(game_state, goalkeeper, penalty_kick=True))
 
         self.assign_tactics(role_to_positions)
 
+    def not_goalkeeper_is_top_or_timeout(self):
+        return not self.goalkeeper_is_top_or_timeout()
+
+    def goalkeeper_is_top_or_timeout(self):
```

Comment:
Shoud be `goalkeeper_is_bot_or_timeout`, since we kick top when the goalkeeper is bot